### DISCUSSION TITLE
How to safely to reduce the times of serializer when input in big content?

### DISCUSSION THREAD

**Discussion by 4-1-1** [Milkdown/milkdown#621] [Q&A] [answered]
When run with big content like one hundred thousand words. 

It will be quite slow , cause to serialize the doc each tr. 
 
Is a way to ensure to finally serialize the doc and callback markdownUpdated after  input  done

**Comment by Saul-Mirone** [comment, marked as answer]
You can call action to get markdown instead of using listener. With action, you can decide when to get content yourself.

**Reply by 4-1-1** [reply]
I just can't decide when  to get the  content . the handler of change and save is not all in control. 😭

**Reply by Saul-Mirone** [reply]
I'll try to optimize the time cost.

**Reply by 4-1-1** [reply]
I decide to callback a function .  to serialize manually when need

**Reply by 4-1-1** [reply]
<img width="631" alt="image" src="https://user-images.githubusercontent.com/4401981/183055768-3b521d83-f179-464e-899a-1e30bf07ab29.png">

**Reply by 4-1-1** [reply]
Found the listener 'updated'. It's a  better way
